beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -1631,10 +1724,11 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                 // network.
                 let seen_clock = &self.chain.slot_clock.freeze_at(seen_timestamp);
                 let hindsight_verification =
-                    attestation_verification::verify_propagation_slot_range(
-                        seen_clock,
-                        failed_att.attestation(),
-                        &self.chain.spec,
+                    attestation_verification::verify_propagation_slot_range::<
+                        ManualSlotClock,
+                        T::EthSpec,
+                    >(
```

Comment:
Maybe we can put the `EthSpec` type parameter first and then elide the slot clock (which should be inferred). Like:

```rust
verify_propagation_slot_range::<T::EthSpec, _>(..)
```